packages/realm/binding/node/CMakeLists.txt

```diff
@@ -0,0 +1,141 @@
+# The initial part of this file is basically just copied from the root package CML files.
+# TODO: look into how to commonize some of this.
+
+include(CheckCXXCompilerFlag)
+
+set(SDK_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../)
+set(BINDGEN_DIR ${SDK_DIR}/bindgen)
+set(BINDING_DIR ${SDK_DIR}/binding)
+
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+    set(CMAKE_SYSTEM_VERSION 8.1)
+    if(CMAKE_GENERATOR_PLATFORM MATCHES "^[Ww][Ii][Nn]32$")
+        set(VCPKG_TARGET_TRIPLET "x86-windows-static" CACHE STRING "")
+    elseif(CMAKE_GENERATOR_PLATFORM MATCHES "^[Xx]64$")
+        set(VCPKG_TARGET_TRIPLET "x64-windows-static" CACHE STRING "")
+    else()
+        message(FATAL_ERROR "Unable to deduce vcpkg target triplet (from '${NODE_ARCH}')")
+    endif()
+elseif(APPLE)
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
+    if(NODE_ARCH STREQUAL "arm64")
+    set(CMAKE_OSX_ARCHITECTURES "arm64" CACHE STRING "")
+    else()
+    set(CMAKE_OSX_ARCHITECTURES "x86_64" CACHE STRING "")
+    endif()
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    if(NODE_ARCH STREQUAL "arm")
+    set(CMAKE_TOOLCHAIN_FILE "${BINDGEN_DIR}/vendor/realm-core/tools/cmake/armv7-linux-gnueabihf.toolchain.cmake")
+    add_link_options(-fuse-ld=bfd)
+    # due to how multiarch works in debian this is needed to link to the correct system libraries
+    set(CMAKE_IGNORE_PATH "/usr/lib/x86_64-linux-gnu")
+    elseif(NODE_ARCH STREQUAL "arm64")
+    set(CMAKE_TOOLCHAIN_FILE "${BINDGEN_DIR}/vendor/realm-core/tools/cmake/aarch64-linux-gnu.toolchain.cmake")
+    set(CMAKE_IGNORE_PATH "/usr/lib/x86_64-linux-gnu")
+    add_link_options(-fuse-ld=gold)
+    add_link_options(-Wl,-z,noexecstack)
+    add_link_options(-Wl,--warn-execstack)
+    elseif(NODE_ARCH STREQUAL "x64")
+    set(CMAKE_TOOLCHAIN_FILE "${BINDGEN_DIR}/vendor/realm-core/tools/cmake/x86_64-linux-gnu.toolchain.cmake")
+    add_link_options(-fuse-ld=gold)
+    add_link_options(-Wl,-z,noexecstack)
+    add_link_options(-Wl,--warn-execstack)
+    endif()
+endif()
+
+cmake_minimum_required(VERSION 3.17)
+project(RealmJS)
+
+# Use ccache if available
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
+endif()
+
+# Copied from Realm Core's CMakeList.txt
+function(add_cxx_flag_if_supported flag)
+    if(flag MATCHES "^-Wno-")
+        # Compilers ignore unknown -Wno-foo flags, so look for -Wfoo instead.
+        string(REPLACE "-Wno-" "-W" check_flag ${flag})
+    else()
+        set(check_flag ${flag})
+    endif()
+
+    check_cxx_compiler_flag(${check_flag} HAVE${check_flag})
+    if(HAVE${check_flag})
+        add_compile_options($<$<COMPILE_LANGUAGE:CXX>:${flag}>)
+    endif()
+endfunction()
+
+set(CMAKE_CXX_STANDARD 20)
```

Comment:
Is there a way to inherit it from Core's CMakeLists.txt?

Reply:
Might be, but technically the Core CMake project is a dependency of ours, so that would be a bit reversed 🤔